src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/SelfHostDeployer.cs

```diff
@@ -189,5 +192,52 @@ public override void Dispose()
                 StopTimer();
             }
         }
+
+        // Temporary workaround for https://github.com/dotnet/cli/issues/6286
+        private void ProvideExecutePermission()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                && DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr
+                && DeploymentParameters.ApplicationType == ApplicationType.Standalone)
+            {
+                var executablePath = Path.Combine(
+                    DeploymentParameters.PublishedApplicationRootPath,
+                    DeploymentParameters.ApplicationName);
+
+                var chmodProcessStartInfo = new ProcessStartInfo
+                {
+                    FileName = "chmod",
+                    Arguments = $"755 {executablePath}",
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+
+                var chmodProcess = new Process() { StartInfo = chmodProcessStartInfo };
+                try
+                {
+                    chmodProcess.StartAndCaptureOutAndErrToLogger("chmod", Logger);
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(
+                        "Error occurred while starting the process '{processName}'. Exception: {exception}",
+                        chmodProcessStartInfo.FileName,
+                        ex.ToString());
+                }
+
+                chmodProcess.WaitForExit(3 * 1000);
+
+                if (chmodProcess.HasExited && chmodProcess.ExitCode != 0)
```

Comment:
Also log if the chmod process did not exit (i.e. hang) though I don't even know if that's even possible.